Tests/MapboxCoreNavigationTests/RouteControllerTests.swift

```diff
@@ -9,538 +9,1224 @@ import MapboxNavigationNative
-            
-            completion(.success(IndexedRouteResponse(routeResponse: routeResponse,
-                                                     routeIndex: 0)))
+        navigatorSpy.returnedSetAlternativeRoutesResult = .success([alternativeRoute])
+        NotificationCenter.default.post(name: .navigatorDidChangeAlternativeRoutes, object: nil, userInfo: userInfo)
+
+        XCTAssertTrue(navigatorSpy.setAlternativeRoutesCalled)
+        XCTAssertEqual(routeController.continuousAlternatives.count, 1)
+        XCTAssertEqual(routeController.continuousAlternatives[0].id, 1)
+        waitForExpectations(timeout: expectationsTimeout)
+    }
+
+    func testSendAlternativeRoutesNotificationIfEmptyCurrentContinuousAlternatives() {
+        expectation(forNotification: .routeControllerDidUpdateAlternatives, object: routeController) { (notification) -> Bool in
+            let userInfo = notification.userInfo
+
+            let updatedAlternatives = userInfo?[RouteController.NotificationUserInfoKey.updatedAlternativesKey] as? [AlternativeRoute]
+            let removedAlternatives = userInfo?[RouteController.NotificationUserInfoKey.removedAlternativesKey] as? [AlternativeRoute]
+            XCTAssertEqual(updatedAlternatives?.count, 1)
+            XCTAssertEqual(removedAlternatives?.count, 0)
+
+            return true
         }
-        
-        let routeController = RouteController(indexedRouteResponse: indexedRouteResponse,
-                                              customRoutingProvider: routingProviderStub,
-                                              dataSource: self)
-        let routerDelegateSpy = RouterDelegateSpy()
-        let routeExpectation = XCTestExpectation(description: "Proactive ReRoute should be called")
-        
-        routerDelegateSpy.onShouldProactivelyRerouteFrom = { _, _ in
-            routeExpectation.fulfill()
+
+        let alternativeRoute = createRouteAlternative(id: 1)
+        let removedAlternativeRoute = createRouteAlternative(id: 2)
+        let userInfo = [
+            Navigator.NotificationUserInfoKey.alternativesListKey: [alternativeRoute],
+            Navigator.NotificationUserInfoKey.removedAlternativesKey: [removedAlternativeRoute]
+        ]
+        navigatorSpy.returnedSetAlternativeRoutesResult = .success([alternativeRoute])
+        NotificationCenter.default.post(name: .navigatorDidChangeAlternativeRoutes, object: nil, userInfo: userInfo)
+
+        waitForExpectations(timeout: expectationsTimeout)
+    }
+
+    func testAlternativeRoutesReportedIfNonEmptyCurrentContinuousAlternatives() {
+        let notification = Notification(name: .navigationDidSwitchToFallbackVersion, object:nil, userInfo: nil)
+        let originalAlternativeRoute = createRouteAlternative(id: 2)
+        navigatorSpy.returnedSetRoutesResult = .success((mainRouteInfo: nil, alternativeRoutes: [originalAlternativeRoute]))
+        routeController.fallbackToOffline(notification)
```

Comment:
We can discuss how to implement it better.

It was my way to set the first value for `continuousAlternatives`. Because the new values of `continuousAlternatives` depends on previously set `alternatives`.
https://github.com/mapbox/mapbox-navigation-ios/blob/86eafe840539e6ad9506160685318c1febd497f8/Sources/MapboxCoreNavigation/RouteController.swift#L317
I changed the tests to use initial call of `setRoutes`, but it is also a questionable decision

Reply:
You could also use `updateRoute(with:)` to set route and alternatives.
BTW, I think we could also craft some tests for fallback to offline and restoring to online feature.

Reply:
I updated tests to use `updateRoute(with:)`.

And I think we cannot do something specific in unit test for offline/online because it's just the call  of `updateNavigator(with:fromLegIndex:completion:)` under the hood, which is tested. 
I think, this case actually is a good candidate for integration tests